### DISCUSSION TITLE
Change Button Address with Potentiometer

### DISCUSSION THREAD

**Discussion by mattibatti** [tttapa/Control-Surface#1017] [General]
Hi - I want to use a potentiometer to change the addresses for my buttons. I see lots of examples using banks to change the CC address of analog inputs, so I tried to use that for buttons by making a button as follows:

```
Bank<8> bank(2,0)

Bankable::CCButton button1 {
  {bank, 0}, 2, {0x07, 0}, 
};
```
But this did not work. What am I missing?

Thanks

**Comment by tttapa** [comment]
There is no real difference between the usage of Bankable::CCButton and Bankable::CCPotentiometer. You can find the details in the [API reference](https://tttapa.github.io/Control-Surface-doc/Doxygen/d1/dd4/classBankable_1_1CCButton.html#aabb58176dc1bc9fccd982bffe0b7eae8). However, it looks like you added multiple invalid arguments (with value 0), what was the intention behind those?

**Comment by mattibatti** [comment]
Awesome, thank you :)

**Comment by mattibatti** [comment]
Hi - I have a question about banks and changing MIDI outputs at runtime. Is
it possible to use banks with custom MIDI outputs for each bank.

For example lets say I have two buttons that i want to control their MIDI
output channels with a switch that has 2 settings. is there a way to use
banks such that instead of each setting incrementing the bank by a certain
amount, it reads off a list of preset custom channels. the end result being
something like

Button1Bank = {1, 2}
Button2Bank = {4,8}

if switch =1{
Button1Bank.(0)
Button2Bank.(0)}
}
if switch = 2{
Button1Bank.(1)
Button2Bank(1)
}

Does that make sense?

Thanks!
m

**Comment by mattibatti** [comment]
Also the same goes for a bankable potentiometer
